src/components/volunteer/VolunteerFormCalendar.tsx

```diff
@@ -70,10 +73,37 @@ export const VolunteerFormCalendar = ({
 
   const isFocused = useIsFocused();
 
-  const { mutate, isLoading, isError, isSuccess, data, reset } = useMutation(calendarNew);
+  const { mutateAsync, isLoading, isError, isSuccess, data, reset } = useMutation(calendarNew);
   const [createEvent] = useMainserverMutation(CREATE_EVENT_RECORDS);
-  const onSubmit = (calendarNewData: VolunteerCalendar) => {
-    mutate(calendarNewData);
+  const onSubmit = async (calendarNewData: VolunteerCalendar) => {
+    mutateAsync(calendarNewData).then(async ({ id }) => {
+      if (id) {
+        const uris = [];
+
+        if (calendarNewData.images) {
+          JSON.parse(calendarNewData.images).forEach(({ uri = String, mimeType = String }) => {
+            uris.push({ uri, mimeType });
+          });
+        }
+
+        if (calendarNewData.documents) {
+          JSON.parse(calendarNewData.documents).forEach(({ uri = String, mimeType = String }) => {
+            uris.push({ uri, mimeType });
+          });
+        }
```

Comment:
please parse the data before checking them. the defaults are `'[]'` strings, so these are always "true":

```
Object {
  "bool": true,
  "images": "[]",
}
```

with parsing them before checking you will have arrays to check length for.

Reply:
now it's safer 👍
https://github.com/ikuseiGmbH/smart-village-app-app/pull/606/commits/a639aac9df2185059b0baa42df9268df5d95b3db

Reply:
imagine parsing fails and `imagesParser` or `documentsParser` would be undefined? then checking `.length` errors.
`...Parser` sounds like a method doing something but these values are arrays representing multiple entries, right? i suggest `images` and `documents` instead to have `uris, images, documents` or if you want to have the info about parsing (if so, tell me why?) then `parsedImages` and `parsedDocuments`, you know?

Reply:
I think it's done now.
https://github.com/ikuseiGmbH/smart-village-app-app/pull/606/commits/9e1e9f647b49b0557f15530961d7e9028cf5c928